_episodes/02-image-basics.md

```diff
@@ -97,54 +60,214 @@ color, but that each pixel can have a different color from its neighbors.
+
+~~~
+#set the random seed so we all get the same matrix
+pseudorandomizer = np.random.RandomState(2021)
+#create a 4 X 4 checkerboard of random colors
+checkerboard = pseudorandomizer.randint(0,255,size=(4,4,3)
+                                       )
+#restore the default map as you show the image
+fig, ax = plt.subplots()
+plt.imshow(checkerboard)
+#display the arrays
+print(checkerboard)
+~~~
+{: .language-python}
+
+![Image of checkerboard matrix](../fig/02-matrix-checkerboard.png)
+![Image of checkerboard](../fig/02-checkerboard.png)
+
+Previously we had one number being mapped to one color or intentsity. Now we are combining the effect of 3 numbers to arrive at a single color value.  Let's see an example of that using the blue square at the end of the second row, which has the index [1,3] 
+
+~~~
+# extract all the color information for the blue square
+upper_right_square = checkerboard[1,3,:]
+upper_right_square
+~~~
+{: .language-python}
+
+This output: array([  7,   1, 110])
+The integers in order represent Red, Green, and Blue.  Looking at the 3 values and knowing how they map, can you understand why it is blue.  If we divide each value by 255, which is the maxium, we can determine how much it is contributing relative to its maximum potential. Effectively, the red is at 7/255 or 2.8 percent of its potential, the green is at 1/255 or 0.4 percent, and blue is 110/255 or 43.1 percent of its potential.  So when you mix those three intensities of color, blue is winning by a wide margin, but the red and green still contribute to make it a slightly different shade of blue than 0,0,110 would be on its own.
+
+These colors mapped to dimensions of the matrix may be referred to as channels.  It may be helpful to understanding if we display each of these channels independently.  We can do that by multiplying our image array representation with a 1d matrix that has a one for the channel we want to keep and zeros for the rest.
+
+~~~
+red_channel = checkerboard * [1,0,0]
+fig, ax = plt.subplots()
+plt.imshow(red_channel)
+~~~
+{: .language-python}
+![Image of red channel](../fig/02-red-channel.png)
+~~~
+green_channel = checkerboard * [0,1,0]
+fig, ax = plt.subplots()
+plt.imshow(green_channel)
+~~~
+{: .language-python}
+![Image of green channel](../fig/02-green-channel.png)
+~~~
+blue_channel = checkerboard * [0,0,1]
+fig, ax = plt.subplots()
+plt.imshow(blue_channel)
+~~~
+{: .language-python}
+![Image of blue channel](../fig/02-blue-channel.png)
+If we look at the upper [1,3] square in all three figures, we can see each of those color contributions in action.  Notice that there are several squares in the blue figure that look even more intensely blue than square [1,3].  When all three channels are combined though, the blue light of those squares is being diluted by the relative strength of red and green being mixed in with them.
+
+
+
```

Comment:
```suggestion
```

Reply:
This one was also a head scratcher for me. It seems like it is empty and got marked as outdated by github.  Do you know how to interpret the outdated label?